geom/src/polyline.rs

```diff
@@ -519,24 +519,32 @@ impl PolyLine {
         Ok(result)
     }
 
-    /// The resulting polygon is manually triangulated and may not have a valid outer Ring (but it
-    /// usually does).
+    /// This produces a `Polygon` with a valid `Ring`. It may crash if this polyline was made with
+    /// `unchecked_new`
     pub fn make_polygons(&self, width: Distance) -> Polygon {
+        let tessellation = self.thicken_tessellation(width);
+        let ring = Ring::deduping_new(tessellation.points.clone())
+            .expect("PolyLine::make_polygons() failed");
+        Polygon::pretessellated(vec![ring], tessellation)
+    }
+
+    /// Just produces a Tessellation
+    pub fn thicken_tessellation(&self, width: Distance) -> Tessellation {
```

Comment:
maybe `fn tessellate(&self, width: Distance)` would be clear enough while being more concise.

Just a naming nit, so feel free to ignore.